### PR TITLE
Skipping ChannelUpdate and ChannelUpdatedByUser events

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Fix the sample app. Leaving a channel has now a problem where the users don't immediately leave the channel. 

Note: There's no need to consider events that may come for channels that are being watched, but the user is not a member. The sample app considers that the user only watches channels where the user is a member. 

### 🛠 Implementation details

Skipping the events `ChannelUpdateEvent` and `ChannelUpdatedByUserEvent` as these events can arrive right after the user left the channel.

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/150133619-189c30eb-70f6-4717-82eb-e27ce8b111a0.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/150133593-82fa4eb8-8358-4ee4-9240-8047d44c95fe.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

Create a channel and leave this channel. Check that this channel is instantly removed from the list.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x PR targets the `develop` branch

#### Code & documentation
- ~[ ] Changelog is updated with client-facing changes~. This PR doesn't affect the SDK.
- ~[] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

**Review by adasiewiczr — COMMENTED** [review]
It fixes leaving a channel but breaks creating distinct channels.
Check the video below:
https://user-images.githubusercontent.com/17440581/150331832-90463e81-d3ad-44c2-a9d8-b13eb8ffe069.mp4
The new channel should be visible after sending a new message but the list is not refreshed properly

**Comment by leandroBorgesFerreira** [comment]
Update: I changed the PR. Now a direct messages channel appears right after its creation, before it was needed to close and reopen the app. 

The bug of many channels created for direct messages is fixed (it is hard to reproduce, so please test it carefully)

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/DefaultChatEventHandler.kt [hunk 1]
```diff
@@ -90,4 +91,16 @@ public class DefaultChatEventHandler(private val channels: StateFlow<List<Channe
         event: NotificationRemovedFromChannelEvent,
         filter: FilterObject,
     ): EventHandlingResult = removeIfChannelIsPresent(channels, event.channel)
+
+    private fun handleChannelUpdate(channel: Channel): EventHandlingResult {
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/DefaultChatEventHandler.kt:95]: I think we shouldn't modify this behavior in our default event handler. Our sample app use case is quite specific.

**Review by adasiewiczr — COMMENTED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [x] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**